indicators.py

```
@@ -635,6 +639,77 @@ def get_all_indicators_signals(self, df: pd.DataFrame) -> Dict[str, any]:
+            
+            return {
+                'signal': signal,
+                'strength': strength,
+                'reason': reason,
+                'volume_ratio': volume_ratio_20,
+                'volume_ratio_50': volume_ratio_50,
+                'volume_spike': volume_spike,
+                'current_volume': current_volume,
+                'avg_volume_20': volume_sma_20,
+                'avg_volume_50': volume_sma_50
+            }
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)